driver-core/src/main/java/com/datastax/driver/core/ParseUtils.java

```diff
@@ -428,45 +460,114 @@ public static String formatTime(long value) {
     }
 
     /**
-     * Simple method to replace a single character.
-     * {@code String.replace()} is a bit too inefficient (see JAVA-67).
-     * This methods is specially useful for escaping single quotes
-     * in CQL strings:
-     * <p/>
-     * <pre>{@code
-     * ParseUtils.replaceChar(value, '\'', "''");
-     * }</pre>
+     * Return {@code true} if the given string is surrounded
+     * by the quote character given, and {@code false} otherwise.
+     *
+     * @param value The string to inspect.
+     * @return {@code true} if the given string is surrounded
+     * by the quote character, and {@code false} otherwise.
+     */
+    private static boolean isQuoted(String value, char quoteChar) {
+        return value != null && value.length() > 1
+                && value.charAt(0) == quoteChar && value.charAt(value.length() - 1) == quoteChar;
+    }
+
+    /**
+     * @param quoteChar " or '
+     * @return A quoted empty string.
+     */
+    private static String emptyQuoted(char quoteChar) {
+        // don't handle non quote characters, this is done so the these are interned and doesn't create
+        // repeated empty quoted strings.
+        assert quoteChar == '"' || quoteChar == '\'';
+        if (quoteChar == '"')
+            return "\"\"";
+        else
+            return "''";
+    }
+
+
+    /**
+     * Quotes text and escapes any existing quotes in the text.
+     * {@code String.replace()} is a bit too inefficient (see JAVA-67, JAVA-1262).
      *
-     * @param text        The text.
-     * @param search      The character to search for.
-     * @param replacement The replacement.
-     * @return The text with all occurrences of {@code search} replaced with {@code replacement}.
+     * @param text      The text.
+     * @param quoteChar The character to use as a quote.
+     * @return The text with surrounded in quotes with all existing quotes escaped with (i.e. ' becomes '')
      */
-    private static String replaceChar(String text, char search, String replacement) {
+    private static String quote(String text, char quoteChar) {
```

Comment:
`quote` and `unquote` are versions of replaceChar with enough differences to justify different methods.  I removed `replaceChar` altogether as it isn't used otherwise, can add it back if you think it is appropriate to keep.